openmdao/visualization/meta_model_viewer/meta_model_visualization.py

```diff
@@ -0,0 +1,554 @@
+        self.slider_value_and_name = OrderedDict()
+        for title, slider_params in self.slider_dict.items():
+            self.slider_value_and_name[title] = slider_params.value
+
+        # Cast the current values of the slider_value_and_name dictionary values to a list
+        self.input_point_list = list(self.slider_value_and_name.values())
+        x_data[:, :, :] = np.array(self.input_point_list)
+
+        # Search the input_data_dict to match the names with the x/y dropdown menus. Then set x/y
+        # linspaces to the values for the meshgrid which follows
+        for title, values in self.input_data_dict.items():
+            if title == self.x_input.value:
+                xlins = values
+                dw = max(values)
+            if title == self.y_input.value:
+                ylins = values
+                dh = max(values)
+
+        # Create a mesh grid and then append that data to 'x_data' in the respective columns
+        X, Y = np.meshgrid(xlins, ylins)
+        x_data[:, :, self.x_index] = X
+        x_data[:, :, self.y_index] = Y
+
+        # This block places the x and y inputs first and then appends any
+        # other values to the list the first two points
+        pred_dict = {}
+        self.input_list = [self.x_input.value, self.y_input.value]
+        for title in self.slider_value_and_name:
+            if title == self.x_input.value or title == self.y_input.value:
+                pass
+            else:
+                self.input_list.append(title)
+
+        # Append the key (input_list) and the values copied from x_data to pred_dict
+        # where it is then ordered in pred_dict_ordered.
+        for idx, title in enumerate(self.slider_value_and_name):
+            pred_dict.update({title: x_data[:, :, idx]})
+        pred_dict_ordered = OrderedDict((k, pred_dict[k]) for k in self.input_list)
+
+        # pred_dict.update(self.slider_value_and_name.items())
+        # pred_dict_ordered = OrderedDict((k, pred_dict[k]) for k in self.input_list)
+
+        # Pass the dict to make predictions and then reshape the output to (n, n, number of outputs)
+        y_data[:, :, :] = self._make_predictions(pred_dict_ordered).reshape(
+            (resolution, resolution, self.num_of_outputs))
+        Z = y_data[:, :, self.output_variable]
+        Z = Z.reshape(resolution, resolution)
+        self.Z = Z
+
+        self.source.add(Z, 'z')
+
+        # Color bar formatting
+        color_mapper = LinearColorMapper(palette="Viridis11", low=np.amin(Z), high=np.amax(Z))
+        color_bar = ColorBar(color_mapper=color_mapper, ticker=BasicTicker(), label_standoff=12,
+                             location=(0, 0))
+
+        # Contour Plot
+        self.contour_plot = figure(
+            tooltips=[(self.x_input.value, "$x"), (self.y_input.value, "$y"),
+                      (self.output_select.value, "@image")], tools="pan")
```

Comment:
could make a local reference here also so you don't have to go through `self.` in the next 10 references

Reply:
This is a `self.` reference because I add to the contour_plot figure in the `bot_plot` and `right_plot` functions

Reply:
sorry, I meant that you could make a temporary local reference, something like:
`self.contour_plot = contour_plot = figure(`
so you wouldn't need to go through `self.` for the remaining references in this function

Reply:
Ah ok. I'll make that change